docs/architecture/adr-009-non-interactive-default-rules-for-reduced-padding.md

```diff
@@ -0,0 +1,177 @@
+# ADR 009: Non-Interactive Default Rules for Reduced Padding
+
+## Changelog
+
+- 14.11.2022: Initial Draft
+
+## Context
+
+[ADR 008](./adr-008-square-size-independent-message-commitments.md) makes it possible for the current non-interactive default rules to be modified slightly:
+
+Current
+
+> - Messages that span multiple rows must begin at the start of a row (this can occur if a message is longer than k shares or if the block producer decides to start a message partway through a row and it cannot fit).
+> - Messages begin at a location aligned with the largest power of 2 that is not larger than the message length or k.
+
+Proposed
+
+> - Messages start at an index that is a multiple of its `msgMinSquareSize`.
+
+The upside of this proposal is that it reduces the inter-message padding. The downside is that a message inclusion proof will not be as efficient for large square sizes so the proof will be larger.
+
+> **Note**  
+> This analysis assumes the implementation of [celestia-app#1004](https://github.com/celestiaorg/celestia-app/issues/1004). If the tree over the subtree roots is not a Namespace Merkle Tree then both methods have the same proof size.
```

Comment:
Even if it's an NMT wouldn't it still have the same size, because the row roots also contain the erasure coded data? Which means you need to "open" each row roots anyways in order to get the non-parity shares.

Reply:
The parity shares only happen at the top of the tree so at least I need to download each row root. If we assume that we have log(n) rows than we still need sqrt(n) subtree roots vs log(n) roots, becasue i am  talking about proving subroots within the same row.